analyze.module

```diff
@@ -25,4 +29,232 @@ function analyze_theme() {
+  if (str_starts_with($route_name, 'analyze.')) {
+    /** @var \Drupal\analyze\AnalyzePluginManager $service */
+    $service = \Drupal::service('plugin.manager.analyze');
+
+    // Add local actions for all entities if we have plugins enabled.
+    if ($plugins = $service->getDefinitions()) {
+      foreach (analyze_get_entity_types() as $id => $route) {
+        foreach ($plugins as $plugin) {
+          $plugin_route = 'analyze.' . $id . '.' . $plugin['id'];
+          $entity_route = 'entity.' . $id . '.canonical';
+
+          if ($route_name === $plugin_route) {
+            $canonical_local_tasks = \Drupal::service('plugin.manager.menu.local_task')
+              ->getLocalTasks($entity_route);
+            $local_tasks['tabs'][0] = $canonical_local_tasks['tabs'];
+
+            // Mark the version history local task as active when viewing a node
+            // revision.
+            foreach ($local_tasks['tabs'][0] as $route_name => &$tab) {
+              $tab['#active'] = $route_name === 'entity.' . $id . '.analyze';
+            }
+
+            $cacheability = CacheableMetadata::createFromObject($cacheability)
+              ->merge($canonical_local_tasks['cacheability']);
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function analyze_form_alter(&$form, FormStateInterface $form_state, $form_id): void {
+  if ($form_id !== 'analyze_analyze_settings') {
+    $is_bundle_form = $entity_type = $bundle = FALSE;
+    $form_object = $form_state->getFormObject();
+
+    // User's aren't bundle-able, but we want the opportunity to include them in
+    // insights so we'll hard-code the form id.
+    if ($form_id == 'user_admin_settings') {
+      $is_bundle_form = TRUE;
+      $entity_type = 'user';
+      $bundle = 'user';
+    }
+    else {
+
+      // Otherwise, we need to identify if this is a bundle entity's edit form
+      // by digging down to get the bundle_of setting.
+      if (method_exists($form_object, 'getEntity') && $entity = $form_object->getEntity()) {
+        if (method_exists($entity, 'getEntityType') && $type = $entity->getEntityType()) {
+          if (method_exists($type, 'getBundleOf') && $entity_type = $type->getBundleOf()) {
+            $is_bundle_form = TRUE;
+            $bundle = $entity->id();
+          }
+        }
+      }
+    }
+
+    if ($is_bundle_form) {
+      analyze_add_settings_form($form, $entity_type, $bundle);
+      $form_state->set('entity_type', $entity_type);
+      $form_state->set('bundle', $bundle);
+    }
+  }
+}
+
+/**
+ * Helper to add Analyze settings to a form.
+ *
+ * @param mixed[] $form
+ *   The form.
+ * @param string $entity_type
+ *   The entity type the settings are for.
+ * @param string $bundle
+ *   The bundle the settings are for.
+ */
+function analyze_add_settings_form(array &$form, string $entity_type, string $bundle): void {
+  if (\Drupal::currentUser()->hasPermission('administer analyze')) {
```

Comment:
I'd generally recommend returning early rather than large conditional blocks, so:
```
if (!\Drupal::currentUser()->hasPermissions('administer analyze')) {
  return;
}

// Do your modifications.
```

Reply:
I think that's a personal preference: I find this easier to read and it prevents issues where people forget that a function has a required return type.